File: android/src/main/java/com/zj/zjsdk/plugin/ZjExpressVideoAdView.java
package com.zj.zjsdk.plugin;

import android.app.Activity;
import android.content.Context;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.zj.zjsdk.ad.ZjAdError;
import com.zj.zjsdk.ad.ZjSize;
import com.zj.zjsdk.ad.express.ZjExpressFeedFullVideo;
import com.zj.zjsdk.ad.express.ZjExpressFeedFullVideoAd;
import com.zj.zjsdk.ad.express.ZjExpressFeedFullVideoListener;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.EventChannel;
import io.flutter.plugin.platform.PlatformView;

public class ZjExpressVideoAdView implements PlatformView {

    private static final String TAG = "ZjExpressVideoAdView";

    private final ViewGroup container;

    private EventChannel.EventSink mEventSink;

    public ZjExpressVideoAdView(Context context, Activity activity, BinaryMessenger messenger, int id, String posId, int width, int height) {
        EventChannel eventChannel = new EventChannel(messenger, "flutter_zjsdk_plugin/express_video_event_" + id);
        eventChannel.setStreamHandler(new EventChannel.StreamHandler() {
            @Override
            public void onListen(Object o, EventChannel.EventSink eventSink) {
                Log.d(TAG, "onListen");
                mEventSink = eventSink;
            }

            @Override
            public void onCancel(Object o) {
                Log.d(TAG, "onCancel");
            }
        });

        container = new FrameLayout(context);
        container.setLayoutParams(new FrameLayout.LayoutParams(ZjUITool.toPx(context, width), ZjUITool.toPx(context, height)));
        loadAd(activity, posId, width, height);
    }

    private void loadAd(Activity activity, String posId, int width, int height) {
        ZjExpressFeedFullVideo zjExpressFullVideoFeed = new ZjExpressFeedFullVideo(activity, posId, new ZjSize(ZjUITool.toPx(activity, width), ZjUITool.toPx(activity, height)), new ZjExpressFeedFullVideoListener() {

            @Override
            public void onZjAdError(ZjAdError zjAdError) {
                Map<String, Object> result = new HashMap<>();
                result.put("event", "onZjAdError");
                result.put("code", zjAdError.getErrorCode());
                result.put("message", zjAdError.getErrorMsg());
                if (mEventSink != null) {
                    mEventSink.success(result);
                    mEventSink.endOfStream();
                }
            }

            @Override
            public void onZjFeedFullVideoLoad(List<ZjExpressFeedFullVideoAd> list) {
                Map<String, Object> result = new HashMap<>();
                result.put("event", "onZjFeedFullVideoLoad");

                if (list != null && list.size() > 0) {
                    result.put("size", String.valueOf(list.size()));
                    ZjExpressFeedFullVideoAd ad = list.get(0);
                    ad.setExpressInteractionListener(new ZjExpressFeedFullVideoAd.FeedFullVideoAdInteractionListener() {
                        @Override
                        public void onAdClicked(View view, int i) {
                            Map<String, Object> result = new HashMap<>();
                            result.put("event", "onZjAdClicked");
                            if (mEventSink != null) {
                                mEventSink.success(result);
                            }
                        }

                        @Override
                        public void onAdShow(View view, int i) {
                            Map<String, Object> result = new HashMap<>();
                            result.put("event", "onZjAdShow");
                            if (mEventSink != null) {
                                mEventSink.success(result);
                            }
                        }

                        @Override
                        public void onRenderSuccess(View view, float v, float v1) {
                            Map<String, Object> result = new HashMap<>();
                            result.put("event", "onRenderSuccess");
                            if (mEventSink != null) {
                                mEventSink.success(result);
                            }
                        }

                        @Override
                        public void onRenderFail(View view, ZjAdError zjAdError) {
                            Map<String, Object> result = new HashMap<>();
                            result.put("event", "onRenderFail");
                            result.put("code", zjAdError.getErrorCode());
                            result.put("message", zjAdError.getErrorMsg());
                            if (mEventSink != null) {
                                mEventSink.success(result);
                                mEventSink.endOfStream();
                            }
                        }
                    });
                    ad.render(container);
                } else {
                    result.put("size", "0");
                }

                if (mEventSink != null) {
                    mEventSink.success(result);
                }
            }
        });
        zjExpressFullVideoFeed.loadAd();
    }

    @Override
    public View getView() {
        return container;
    }

    @Override
    public void dispose() {
        Log.e(TAG, "dispose");
    }

}

File: lib/zj_express_video_ad_view.dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

import 'zj_callback.dart';

/// Widget for ZjExpressFeedFullVideo
class ZjExpressVideoAdView extends StatelessWidget {
  /// 广告位ID
  final String posId;

  /// 宽
  final double width;

  /// 高
  final double height;

  /// 加载成功
  final AdCallback1? onZjFeedFullVideoLoad;

  /// 加载失败
  final AdErrorCallback? onZjAdError;

  /// 渲染成功
  final AdCallback0? onRenderSuccess;

  /// 渲染失败
  final AdErrorCallback? onRenderFail;

  /// 广告展示成功
  final AdCallback0? onZjAdShow;

  /// 广告点击
  final AdCallback0? onZjAdClicked;

  ZjExpressVideoAdView(
      {Key? key,
      required this.posId,
      required this.width,
      required this.height,
      this.onZjFeedFullVideoLoad,
      this.onZjAdError,
      this.onRenderSuccess,
      this.onRenderFail,
      this.onZjAdShow,
      this.onZjAdClicked})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    Widget expressVideo;
    if (defaultTargetPlatform == TargetPlatform.android) {
      expressVideo = AndroidView(
        viewType: 'flutter_zjsdk_plugin/express_video',
        creationParams: {
          "posId": posId,
          "width": width,
          "height": height,
        },
        creationParamsCodec: const StandardMessageCodec(),
        onPlatformViewCreated: _onPlatformViewCreated,
      );
    } else {
      expressVideo = Text("Not supported");
    }

    return SafeArea(
      child: Stack(
        children: <Widget>[
          Container(
            width: width,
            height: height,
            child: expressVideo,
          ),
        ],
      ),
    );
  }

  void _onPlatformViewCreated(int id) {
    EventChannel eventChannel =
        EventChannel("flutter_zjsdk_plugin/express_video_event_$id");
    eventChannel.receiveBroadcastStream().listen((event) {
      switch (event["event"]) {
        case "onZjFeedFullVideoLoad":
          print("ZjExpressFeedFullVideoAd.onZjFeedFullVideoLoad");
          onZjFeedFullVideoLoad?.call(event["size"]);
          break;

        case "onZjAdError":
          print(
              "ZjExpressFeedFullVideoAd.onZjAdError[${event["code"]}-${event["message"]}");
          onZjAdError?.call(event["code"], event["message"]);
          break;

        case "onRenderSuccess":
          print("ZjExpressFeedFullVideoAd.onRenderSuccess");
          onRenderSuccess?.call();
          break;

        case "onRenderFail":
          print(
              "ZjExpressFeedFullVideoAd.onRenderFail[${event["code"]}-${event["message"]}");
          onRenderFail?.call(event["code"], event["message"]);
          break;

        case "onZjAdShow":
          print("ZjExpressFeedFullVideoAd.onZjAdShow");
          onZjAdShow?.call();
          break;

        case "onZjAdClicked":
          print("ZjExpressFeedFullVideoAd.onZjAdClicked");
          onZjAdClicked?.call();
          break;
      }
    });
  }
}
